clusters.cnp.example.org/definition.yaml

```diff
@@ -26,10 +36,10 @@ spec:
     group: cnp.example.org
     version: v1alpha1
     names:
-      kind: Cluster
-      listKind: ClusterList
-      plural: clusters
-      singular: cluster
+      kind: AllocatedCluster
+      listKind: AllocatedClusterList
+      plural: allocatedclusters
+      singular: allocatedcluster
```

Comment:
The platform builder gets to pick the name of both the composite resource and the published resource, so they can pick which one has a differentiating suffix or prefix. This would allow the builder to choose between:

* `ClusterClaim` for the published type and `Cluster` for the composite type.
* `ClusterRequirement` for the published type and `Cluster` for the composite type.
* `Cluster` for the published type and `AllocatedCluster` for the composite type.
* `Cluster` for the published type and `CompositeCluster` for the composite type.

The only limitation is that both types can't share the same name - they can't both be `Cluster`.

Reply:
definitely prefer this approach 👍